src/comm/UDPLink.cc

```diff
@@ -182,7 +182,10 @@ void UDPLink::readBytes()
         QHostAddress sender;
         quint16 senderPort;
         //-- Note: This call is broken in Qt 5.9.3 on Windows. It always returns a blank sender and 0 for the port.
-        _socket->readDatagram(datagram.data(), datagram.size(), &sender, &senderPort);
```

Comment:
Very strange. Can you include a comment on the `if -1` explaining why it is needed. I also assume the comment about 5.9.3 is no longer needed. Can you remove that?

Reply:
I print out the error message of the socket: "Connection reset by peer".

<img width="908" alt="Screen Shot 2020-11-02 at 16 46 46" src="https://user-images.githubusercontent.com/5885946/97847768-13ec8900-1d2b-11eb-9a26-7e30c9398464.png">

And the message "Connection reset by peer" will keep pop up until manually "Disconected" the link on QGC side.

Figure out the error code is WSAECONNRESET on Windows, as stated in [windows-sockets-error-code](https://docs.microsoft.com/en-us/windows/win32/winsock/windows-sockets-error-codes-2):

> An existing connection was forcibly closed by the remote host. This normally results if the peer application on the remote host is suddenly stopped, the host is rebooted, the host or remote network interface is disabled, or the remote host uses a hard close.

And in our scenario, the peer application which is PX4 is closed. I think it may not close the link properly(NOT verified). Since we can't control the behavior of other application, condition checking on our-side return value(-1) is required.

Reply:
Ok, so comment should say something along those lines. "If the other end is reset then it will still report data available, but will fail on the readDatagram call". Something like that.

Reply:
adding comments as request.